### PR TITLE
Avx512cmp

### PR DESCRIPTION
**Pull request by jtlap**
masked compares

### CODE REVIEW

**Review by DenisYaroshevskiy — COMMENTED** [review]
I'm not against this but I have to ask why.
What would be the use-case.

**Comment by jtlap** [comment]
frankly the use case is not clear but avx512 has the intrinsics that do the masking with the same lattency as the mere predicate,
so I braugth the interface to eve as we have this for many other function objects.

**Comment by jtlap** [comment]
I have forgotten to add the doc, I'm on it.

File: include/eve/function/is_less.hpp [hunk 1]
```diff
@@ -9,6 +9,8 @@
 
 #include <eve/detail/implementation.hpp>
 #include <eve/traits/as_logical.hpp>
+#include <eve/constant/false.hpp>
```
Comment (jfalcou) [inline include/eve/function/is_less.hpp:12]: why those includes ?

**Review by jfalcou — CHANGES_REQUESTED** [review]

File: include/eve/function/is_less.hpp (hunk 1 above)
Comment (jtlap) [reply include/eve/function/is_less.hpp:12]: suppressed

File: include/eve/detail/function/conditional.hpp [hunk 2]
```diff
@@ -60,4 +61,36 @@ namespace eve::detail
                                                                           , alternative(c,a0,eve::as<r_t>{})
                                                                           );
   }
+
+  //================================================================================================
+  // What's the logical alternative for this case ?
+  //================================================================================================
+  template<conditional_expr C, typename Target>
+  EVE_FORCEINLINE auto lalternative(C const& c, as<Target> const&)
```
[the remark this answers is not in the document]
Comment (jtlap) [reply include/eve/detail/function/conditional.hpp]: done

**Review by jfalcou — APPROVED** [review]

**Pull request merged**